homeassistant/components/holiday/calendar.py

```diff
@@ -0,0 +1,120 @@
+"""Holiday Calendar."""
+from __future__ import annotations
+
+from datetime import datetime
+
+from holidays import country_holidays
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_COUNTRY
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from .const import CONF_PROVINCE, DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Holiday Calendar config entry."""
+    country: str = config_entry.data[CONF_COUNTRY]
+    province: str | None = config_entry.data.get(CONF_PROVINCE)
+
+    async_add_entities(
+        [
+            HolidayCalendarEntity(
+                config_entry.title, country, province, unique_id=config_entry.entry_id
+            )
+        ],
+        True,
+    )
+
+
+class HolidayCalendarEntity(CalendarEntity):
+    """Representation of a Holiday Calendar element."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        name: str,
+        country: str,
+        province: str | None,
+        unique_id: str,
+    ) -> None:
+        """Initialize HolidayCalendarEntity."""
+        self._country = country
+        self._province = province
+        self._location = name
+        self._event: CalendarEvent | None = None
+        self._attr_name = None
+        self._attr_unique_id = unique_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unique_id)},
+            name=name,
+        )
+
+    def _default_language(self, hass: HomeAssistant) -> str:
+        """Return the default language."""
+        return (
+            country_holidays(hass.config.country).default_language
+            or hass.config.language
+            if hass.config.country
+            else hass.config.language
+        )
+
+    @property
+    def event(self) -> CalendarEvent | None:
+        """Return the next upcoming event."""
+        obj_holidays = country_holidays(
```

Comment:
Similar to `workday` we could initialize this in `__init__` so we have it and not every pass here (and below)?

Reply:
Sorry, there are so many things new to me. How could I change the years range (below) if I initialize it in `__init__`?

Reply:
I was more thinking if we load it with the current and next year we can presume they will restart HA during that time.

Reply:
If you use the calendar of the frontend you won't see any holidays in the upcoming years then.

Reply:
Yeah. Wasn't sure this was any good and didn't consider the other function really so you're right so we'll skip this one.

Reply:
I don't think it's expensive. Anyway, country_holiday loads an entire year of holidays by default (in our case two because of future events), so it should not make a huge difference.

Reply:
I changed my mind too. Still think we should do this on `__init__` and perhaps save the year or something as a check so we can update it if needed.

Reply:
I moved it to `async_setup_entry`, and update the object if needed.